java/pgjdbc_using_dsql_connector/src/main/java/com/amazon/dsql/samples/CustomCredentialsProviderExample.java

```diff
@@ -0,0 +1,191 @@
+                executeSimpleQuery(connection);
+            }
+            
+        } catch (Exception e) {
+            System.err.println("✗ Profile credentials connection failed: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    /**
+     * Connect using STS Assume Role credentials
+     */
+    private static void connectWithAssumeRoleCredentials() {
+        System.out.println("\n=== Example 2: STS Assume Role Credentials Provider ===");
+        
+        try {
+            // Verify environment variables are set
+            assert CLUSTER_ENDPOINT != null : "CLUSTER_ENDPOINT environment variable is not set";
+            assert CLUSTER_USER != null : "CLUSTER_USER environment variable is not set";
+            
+            if (ROLE_ARN == null || ROLE_ARN.isEmpty()) {
+                System.out.println("Skipping assume role example - ROLE_ARN environment variable not set");
+                return;
+            }
+            
+            System.out.println("Using role ARN: " + ROLE_ARN);
+            
+            // Create STS client - extract region from cluster endpoint
+            String region = extractRegionFromEndpoint(CLUSTER_ENDPOINT);
+            StsClient stsClient = StsClient.builder()
+                    .region(Region.of(region))
+                    .build();
+            
+            // Create assume role credentials provider
+            AwsCredentialsProvider assumeRoleCredentials = StsAssumeRoleCredentialsProvider.builder()
+                    .refreshRequest(
+                            AssumeRoleRequest.builder()
+                                    .roleArn(ROLE_ARN)
+                                    .roleSessionName("aurora-dsql-session")
+                                    .durationSeconds(3600) // 1 hour
+                                    .build()
+                    )
+                    .stsClient(stsClient)
+                    .build();
+            
+            // Set the custom credentials provider
+            AuroraDsqlCredentialsManager.setProvider(assumeRoleCredentials);
+            
+            // Connect to Aurora DSQL using URL
+            String url = "jdbc:aws-dsql:postgresql://" + CLUSTER_ENDPOINT + "/postgres?user=" + CLUSTER_USER;
+            
+            try (Connection connection = DriverManager.getConnection(url)) {
+                System.out.println("✓ Connected using assume role credentials");
+                executeSimpleQuery(connection);
+            }
+            
+        } catch (Exception e) {
+            System.err.println("✗ Assume role credentials connection failed: " + e.getMessage());
+            e.printStackTrace();
+        }
+    }
+    
+    /**
+     * Extract region from Aurora DSQL endpoint
+     */
+    private static String extractRegionFromEndpoint(String endpoint) {
+        if (endpoint == null || endpoint.isEmpty()) {
+            return "us-east-1"; // default fallback
```

Comment:
It seems like it could create some pretty confusing errors for users if the fallback ends up being used here. Given the limited scope of this sample, I think we should fail fast here and throw a clear error if the region can't be parsed given it should always be possible.

The other samples provide a `REGION` environment variable so that could also be added here as a fallback.